src/topology/builder.rs

```diff
@@ -287,6 +314,10 @@ pub async fn build_pieces(
         detach_triggers.insert(id.clone(), trigger);
     }
 
+    // We should have all the data for the enrichment tables loaded now, so switch them over to
+    // readonly.
+    context.enrichment_tables.finish_load();
```

Comment:
I'm still struggling to tell if we're handling reloads properly. From what I can tell, we build tables that are present as "added" in the config diff, then we create a new `Tables` and pass it to a new `TransformContext`, which then gets passed to newly built transforms. Which leaves me with a couple questions:

1. Can a newly added transform access an existing enrichment table?
2. Do we ever drop tables that are removed from the config?

I get that we're not managing these individually and will have to make some tradeoffs, but it seems like there may be some gaps here. Then again, reloads are confusing and I could be misunderstanding.